src/tests/ftest/telemetry/telemetry_pool_metrics.py

```diff
@@ -22,163 +20,155 @@ class TelemetryPoolMetrics(IorTestBase, TestWithTelemetry):
 
         :avocado: tags=all,daily_regression
         :avocado: tags=vm
         :avocado: tags=telemetry
         :avocado: tags=telemetry_pool_metrics,test_telemetry_pool_metrics
         """
-        # test parameters
-        self.threshold_percent = self.params.get("threshold_percent",
-                                                 "/run/telemetry_metrics/*")
-        metric_list = ["engine_pool_ops_fetch",
-                       "engine_pool_ops_update",
-                       "engine_pool_xferred_fetch",
-                       "engine_pool_xferred_update"]
-        metrics_data = {}
+
         # create pool and container
-        idx = 0
         self.add_pool(connect=False)
         self.pool.set_property("reclaim", "disabled")
         self.add_container(pool=self.pool)
+
         # collect first set of pool metric data before read/write
-        metrics_data[idx] = self.telemetry.get_pool_metrics(metric_list)
-        idx += 1
+        metric_names = [
+            "engine_pool_ops_fetch",
+            "engine_pool_ops_update",
+            "engine_pool_xferred_fetch",
+            "engine_pool_xferred_update",
+            "engine_pool_resent"
+        ]
+        metrics_data = []
+        metrics_data.append(self.telemetry.get_pool_metrics(metric_names))
 
         # Run ior command.
         try:
             self.update_ior_cmd_with_pool(False)
-            self.ior_cmd.dfs_oclass.update(self.container.oclass.value)
+            self.ior_cmd.dfs_oclass.update(self.dfs_oclass)
+            self.ior_cmd.transfer_size.update(self.ior_transfer_size)
+            self.ior_cmd.dfs_chunk.update(self.ior_transfer_size)
+            self.ior_cmd.block_size.update(self.ior_block_size)
             self.run_ior_with_pool(
                 timeout=200, create_pool=False, create_cont=False)
         except TestFail:
             self.log.info("#ior command failed!")
+
         # collect second set of pool metric data after read/write
-        metrics_data[idx] = self.telemetry.get_pool_metrics(metric_list)
-        # collect data for expected values
-        expected_total_objects = (self.ior_cmd.block_size.value /
-                                  self.ior_cmd.dfs_chunk.value) + 1
-        #     Number of expected total objects, Chunk Size
-        check_values = [expected_total_objects, self.ior_cmd.dfs_chunk.value]
+        metrics_data.append(self.telemetry.get_pool_metrics(metric_names))
+
+        metric_values = {}
+        for name in metric_names:
+            metric_values[name] = self.get_metric_value(name, metrics_data)
+
         # perform verification check
-        self.verify_pool_metrics(metric_list, metrics_data, check_values)
+        expected_values = self.get_expected_values(metric_values["engine_pool_resent"])
+        for name in expected_values:
+            val = metric_values[name]
+            min_val = expected_values[name][0]
+            max_val = expected_values[name][1]
```

Comment:
Just FYI this works in Python as long as the list/tuple has length 2
```suggestion
            min_val, max_val = expected_values[name]
```

Reply:
- [ ] Use pair assignment